src/main/java/io/prometheus/cloudwatch/WebServer.java

```diff
@@ -2,27 +2,41 @@
 
 import io.prometheus.client.exporter.MetricsServlet;
 import java.io.FileReader;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 import org.eclipse.jetty.servlet.ServletHolder;
+import sun.misc.Signal;
+import sun.misc.SignalHandler;
 
 public class WebServer {
-   public static void main(String[] args) throws Exception {
-     if (args.length < 2) {
-       System.err.println("Usage: WebServer <port> <yml configuration file>");
-       System.exit(1);
-     }
-     CloudWatchCollector cc = new CloudWatchCollector(new FileReader(args[1])).register();
-
-     int port = Integer.parseInt(args[0]);
-     Server server = new Server(port);
-     ServletContextHandler context = new ServletContextHandler();
-     context.setContextPath("/");
-     server.setHandler(context);
-     context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
-     context.addServlet(new ServletHolder(new DynamicReloadServlet(cc, args[1])), "/-/reload");
-     context.addServlet(new ServletHolder(new HomePageServlet()), "/");
-     server.start();
-     server.join();
-   }
+
+    public static String configFilePath;
+    public static CloudWatchCollector collector;
+
+    public static void main(String[] args) throws Exception {
+        if (args.length < 2) {
+            System.err.println("Usage: WebServer <port> <yml configuration file>");
+            System.exit(1);
+        }
+
+        configFilePath = args[1];
+        collector = new CloudWatchCollector(new FileReader(configFilePath)).register();
+
+        ReloadSignalHandler.start();
```

Comment:
I'd inline this rather than using public static variables.

Reply:
We can do that, but we'd just be moving the `collector` var from `WebServer` to somewhere else (like `ReloadSignalHandler`) since we need to reference it from both the signal handler and `DynamicReloadServlet`.

Where are you thinking we should store this value, then?

Reply:
I'd pass the collector into the handler and servlet.

Reply:
Gotcha--I misread what you said.  Changed in a014d06.